code/game/objects/items/wayfinding.dm

```diff
@@ -1,32 +1,141 @@
 /obj/machinery/pinpointer_dispenser
-	name = "wayfinding pinpointer dispenser"
-	icon = 'icons/obj/bureaucracy.dmi'
-	icon_state = "ticketmachine"
-	desc = "Having trouble finding your way? This machine dispenses pinpointers that point to common locations."
+	name = "wayfinding pinpointer synthesizer"
+	icon = 'icons/obj/machines/wayfinding.dmi'
+	icon_state = "pinpointersynth"
+	desc = "Having trouble finding your way? This machine synthesizes pinpointers that point to common locations."
 	density = FALSE
 	layer = HIGH_OBJ_LAYER
-	var/list/obj/item/pinpointer/wayfinding/pinpointers = list()
-	var/spawn_cooldown = 1200 //deciseconds per person to spawn another pinpointer
+	var/list/user_spawn_cooldowns = list()
+	var/list/user_interact_cooldowns = list()
+	var/spawn_cooldown = 3000 //deciseconds per person to spawn another pinpointer
+	var/interact_cooldown = 200 //deciseconds per person for subsequent interactions
+	var/start_bal = 200 //how much money it starts with to cover refunds
+	var/datum/bank_account/synth_acc = new /datum/bank_account/wayfinding
+	var/ppt_cost = 65 //Jan 6 '20: Assistant can buy one roundstart (125 cr starting)
+	var/refund_amt = 40 //how much recycling a pinpointer rewards you
+	var/expression_timer
+
+/obj/machinery/pinpointer_dispenser/Initialize(mapload)
+	..()
+	var/datum/bank_account/civ_acc = SSeconomy.get_dep_account(ACCOUNT_CIV)
+	if(civ_acc)
+		synth_acc.transfer_money(civ_acc, start_bal) //float has to come from somewhere, right?
+
+	synth_acc.account_holder = src.name
+
+	desc += " Only [ppt_cost] credits!"
+
+	set_expression("neutral")
 
 /obj/machinery/pinpointer_dispenser/attack_hand(mob/living/carbon/user)
-	if(world.time < pinpointers[user.real_name])
-		var/secsleft = (pinpointers[user.real_name] - world.time) / 10
-		to_chat(user, "<span class='warning'>You need to wait [secsleft/60 > 1 ? "[round(secsleft/60)] minute\s" : "[round(secsleft)] second\s"].</span>")
+	if(world.time < user_interact_cooldowns[user.real_name])
+		to_chat(user, "<span class='warning'>It doesn't respond.</span>")
 		return
 
-	to_chat(user, "<span class='notice'>You take a pinpointer from [src].</span>")
+	user_interact_cooldowns[user.real_name] = world.time + interact_cooldown
+
+	for(var/obj/item/pinpointer/wayfinding/WP in user.GetAllContents())
+		set_expression("unsure", 20)
+		say("<span class='robot'>I know you already have a pinpointer, [user.first_name()].</span>")
+		user_spawn_cooldowns[user.real_name] = world.time + spawn_cooldown //spawn timer resets for trickers
+		return
+
+	var/msg1 = ""
+	var/msg2 = ""
+	var/msg3 = ""
+	var/msg4 = ""
```

Comment:
This is pretty messy. msg1 is never assigned, 2 and 3 there is no reason you cant do in a single var. 
Preferably, do all of this in a single var and just append to it as needed. So replace check thing -> add to msg2/3 var with check thing, add to msg var and do it all in the right order.